posts/posts-router.js

```diff
@@ -0,0 +1,118 @@
+const express = require('express');
+
+const Hubs = require('./db.js');
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+    try {
+        const hubs = await Hubs.find();
+        res.status(200).json(hubs);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The posts information could not be retrieved.',
+        });
+    }
+});
+
+
+router.get('/:id', async (req, res) => {
+    try {
+      const hub = await Hubs.findById(req.params.id);
+  
+      if (hub) {
+        res.status(200).json(hub);
+      } else {
+        res.status(404).json({ message: 'Hub not found' });
+      }
+    } catch (error) {
+      // log error to database
+      console.log(error);
+      res.status(500).json({
+        message: 'The post with the specified ID does not exist.',
+      });
+    }
+  });
+  
+  router.get('/:id/comments', (req, res) => {
+    const { id } = req.params;
+  
+    Hubs.findHubMessages(id)
+      .then(messages => {
+        if (messages.length > 0) {
+          res.status(200).json(messages);
+        } else {
+          res.status(404).json({ message: 'The post with the specified ID does not exist.' });
+        }
+      })
+      .catch(err => {
+        res.status(500).json({
+          message: 'Error retrieving the hub',
+        });
+      })
+  })
+  
+  router.post('/:id/comments', async (req, res) => {
+      const messageInfo = { ...req.body, hub_id: req.params.id };
+      try {
+        const message = await Hubs.addMessage(messageInfo);
+        res.status(201).json(message);
+      } catch (error) {
+        console.log(error);
+        res.status(404).json({
+          message: 'The post with the specified ID does not exist.',
+        });
+      }
+    });
+  
+  router.post('/', async (req, res) => {
+    try {
+      const hub = await Hubs.add(req.body);
+      if (title.length > 0 && body.length > 0) {
```

Comment:
I don't see here where are title and body coming from, you should be getting an error of cannot read property length of undefined when hitting this endpoint.